### PR TITLE
[gvar] Make iteration lazy as well

### PR DESCRIPTION
**Pull request by behdad**
This is continuation of #2741, making iteration lazy-load one item at a time as well.

Tests pass, but more testing would be good, in real applications.

### CODE REVIEW

**Comment by justvanrossum** [comment]
By now I wonder if it isn't better to subclass `collections.abc.MutableMapping`. It only requires you to implement `__getitem__`, `__setitem__`, `__delitem__`, `__iter__`, and `__len__`, and everything else is done for you.

File: Lib/fontTools/ttLib/tables/_g_v_a_r.py [hunk 1]
```diff
@@ -54,14 +54,24 @@ def __getitem__(self, k):
         return v
 
     def items(self):
-        if not hasattr(self, '_loaded'):
-            self._load()
-        return super().items()
+        gen = super().items()
+        def wrap(gen):
+            for v in gen:
```
Comment (justvanrossum) [inline Lib/fontTools/ttLib/tables/_g_v_a_r.py:59]: I would write `for k, v in gen` and aviod the noisy indexing below.

**Comment by justvanrossum** [comment]
Or `collections.UserDict`: perhaps you can get away with only implementing `__getitem__`.

**Comment by justvanrossum** [comment]
I made #2781 with UserDict so you can see.

**Comment by anthrotype** [comment]
I think I also prefer #2781 because the less code the better

**Pull request closed without merging**